src/pages/transactions/[id].js

```diff
@@ -1,11 +1,19 @@
 import { useRouter } from "next/router";
-import transactions from '../../data/transactions.json';
+import transactions from "../../data/transactions.json";
```

Comment:
A minor point: you are currently mixing import paths, some giving relative paths ("../../...."), some starting at root (@). Try to make this consistent, using root-based imports only. Nothing urgent, but have an eye on this.